### PR TITLE
container/list: simplified using core.ListForEachElement()

### PR DESCRIPTION
**Pull request by amery**
## Summary by CodeRabbit

- **New Features**
	- Introduced new utility methods for enhanced element matching and iteration in the list functionality.
	- Improved the `ForEach` method for better readability and efficiency.
	- Streamlined element deletion and retrieval processes with new helper methods.

- **Bug Fixes**
	- Enhanced the logic for retrieving first and last matching elements, improving overall performance.

- **Documentation**
	- Updated dependencies to ensure compatibility and functionality of the module.

### CODE REVIEW

File: container/list/list.go [hunk 1]
```diff
@@ -218,3 +188,52 @@ func (l *List[T]) Copy(fn func(T) (T, bool)) *List[T] {
...
+
+	return match, out, match != nil
+}
+
+func (l *List[T]) unsafeGetAllMatchElement(fn func(T) bool) []*list.Element {
+	var out []*list.Element
+	cb := func(el *list.Element, v T) bool {
+		if fn != nil || fn(v) {
+			out = append(out, el)
+		}
```
[the remark this answers is not in the document]
Comment (amery) [reply container/list/list.go:215]: oops

**Comment by amery** [comment]
@karasz do you agree with the bot that having two functions is better than a backward flag in this case?

**Comment by karasz** [comment]
revive also dislikes flag parameters, I am on the fence with this but inclining on write two functions and avoid using a flag parameter.

**Comment by amery** [comment]
refactored dropping the `backward` flag.

File: container/list/list.go [hunk 2]
```diff
@@ -218,3 +188,94 @@ func (l *List[T]) Copy(fn func(T) (T, bool)) *List[T] {
...
+	}
+
+	cb := func(el *list.Element, v T) bool {
+		if cond == nil || cond(v) {
+			*match = el
+			*out = v
+		}
+
+		return match == nil
+	}
```
[the remark this answers is not in the document]
Comment (amery) [reply container/list/list.go:264]: good catch

[the remark this answers is not in the document]
Comment (amery) [reply container/list/list.go]: amended

**Pull request merged**